### DISCUSSION TITLE
Extract form ChatCompletion - Error

### DISCUSSION THREAD

**Discussion by Art77hur** [openai/openai-python#1150] [Q&A]
I need help : a fresh eye on the code !
I really feel like I'm going in circles to resolve this error

# try:
            chat_completion = client.chat.completions.create(
                model="gpt-3.5-turbo",
                messages=[
                    {"role": "user", "content": prompt}
                ],
                temperature=0.7,
                max_tokens=150,
                top_p=1.0,
                frequency_penalty=0.0,
                presence_penalty=0.0
            )

# chat_completion
> 2024-02-13 - INFO - ChatCompletion(id='chatcmpl-8rlh7Hj4auUxtL9mqQvcXwv0evKG', choices=[Choice(finish_reason='stop', index=0, logprobs=None, message=ChatCompletionMessage(content="**what i want to extract**", role='assistant', function_call=None, tool_calls=None))], created=1707825413, model='gpt-3.5-turbo-0613', object='chat.completion', system_fingerprint=None, usage=CompletionUsage(completion_tokens=53, prompt_tokens=25, total_tokens=78))

# Extraction de la réponse
            generated_text = chat_completion.choices[0].message.content.strip()

# Error
 Erreur lors de la génération du prompt 1: 'ChatCompletion' object is not subscriptable

**Comment by judahpaul16** [comment]
I believe you want to extract the text using `['choices'][0]['text']` for v1/completions. I'm using `openai==0.28.0` on my [project](https://github.com/judahpaul16/gpt-home).

```python
async def query_openai(text, display, retries=3):
    # Load settings from settings.json
    settings = load_settings()
    max_tokens = settings.get("max_tokens")
    temperature = settings.get("temperature")

    for i in range(retries):
        try:
            response = openai.ChatCompletion.create(
                model=settings.get("model"),
                messages=[
                    {"role": "system", "content": f"You are a helpful assistant. {settings.get('custom_instructions')}"},
                    {"role": "user", "content": f"Human: {text}\nAI:"}
                ],
                max_tokens=max_tokens,
                temperature=temperature
            )
            response_content = response['choices'][0]['message']['content'].strip()
            if response_content:  # Check if the response is not empty
                message = response_content
                return message
            else:
                logger.warning(f"Retry {i+1}: Received empty response from OpenAI.")
        except Exception as e:
            if 'Did you mean to use v1/completions?' in str(e):
                # Re-query using v1/completions
                prompt = f"You are a helpful assistant. {settings.get('custom_instructions')}\nHuman: {text}"
                response = openai.Completion.create(
                    model=settings.get("model"),
                    prompt=prompt,
                    max_tokens=max_tokens,
                    temperature=temperature
                )
                response_content = response['choices'][0]['text'].strip()
                if response_content:
                    message = response_content
                    return message
            else:
                logger.error(f"Error on try {i+1}: {e}")
            if i == retries - 1:  # If this was the last retry
                error_message = f"Something went wrong after {retries} retries: {e}"
                await handle_error(error_message, None, display)
        await asyncio.sleep(0.5)  # Wait before retrying
```
